src/com/podevs/android/poAndroid/teambuilder/MoveChooserFragment.java

```diff
@@ -103,25 +103,30 @@ public boolean onItemLongClick(AdapterView<?> parent, View view, int position, l
 		return v;
 	}
 
-	protected void buildHiddenPowerDialog() {
+	protected void buildHiddenPowerDialog(MoveListAdapter moveButton, View view) {
 		final AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
 		builder.setTitle(R.string.hiddenpower_choice)
 			.setSingleChoiceItems(R.array.hp_array, poke().hiddenPowerType()-1, null)
 			.setPositiveButton(R.string.ok, new OnClickListener() {
 				public void onClick(DialogInterface dialog, int which) {
 					ListView lw = ((AlertDialog)dialog).getListView();
 					int type = lw.getCheckedItemPosition() + 1;
-
-					byte[] config = HiddenPowerInfo.configurationForType(type, poke().gen);
-
 					if (poke().gen().num < 7) {
+						byte[] config = HiddenPowerInfo.configurationForType(type, poke().gen);
 						if (config != null) {
 							poke().DVs = config;
-							if (listener != null) {
-								listener.onMovesetChanged(true);
-							}
 						}
 					}
+					else if (poke().validHiddenPowerType(type)) {
+						poke().hiddenPowerType = (byte)type;
+					} else {
+						type = poke().hiddenPowerType();
+					}
+
+					if (listener != null) {
+						listener.onMovesetChanged(true);
+					}
+					moveButton.reloadImage(type, view);
```

Comment:
there is already a function that will achieve this: moveAdapter.notifyDataSetChanged();